### PR TITLE
chore: add typing to home page

### PR DESCRIPTION
**Pull request by BryanPan342**
:) fixed ur typing thing

also.. ngl NOT a fan of husky HAHA 

Also why are you using both eslint and prettier?

### CODE REVIEW

File: pages/index.tsx [hunk 1]
```diff
@@ -1,122 +1,136 @@
...
         </div>
       </div>
-    </div>
-  </Layout>
+    </Layout>
   )
 }
 
-export default Home
+export const getStaticProps: GetStaticProps<HomeProps> = async () => {
```
Comment (BryanPan342) [inline pages/index.tsx:115]: this is all i really added!

File: components/GitHubEvent.tsx [hunk 2]
```diff
@@ -1,48 +1,54 @@
...
 // see https://github.com/octokit/types.ts
 // and https://docs.github.com/en/rest/reference/activity#list-public-events
 // and https://docs.github.com/en/developers/webhooks-and-events/github-event-types
 interface GitHubEvent {
-  type: string,
-  actor: GitHubActor,
-  repo: GitHubRepo,
-  created_at: string,
-  payload: any,
+  id: string
```
Comment (BryanPan342) [inline components/GitHubEvent.tsx:11]: added this as well!

**Comment by BryanPan342** [comment]
@mattxwang looks a lot different now bc of the precommit hook 🥲

I removed the `export default Home` from the bottom of the file to do little more of a `next.js`-ey approach! Usually when I look at other repos, I see something along the lines of 

```ts
interface PageProps() {
}

export default function Page(props: PageProps) {
}

// for dynamic routing
export const getStaticPaths = () => {
}

export const getStaticProps = () => {
}
```

**Review by mattxwang — APPROVED** [review]
Merci buckets friend, looks good to me! Can't believe I'm honoured by a bpan collab 🥺 

> also.. ngl NOT a fan of husky HAHA

LOL I think it's a bit bloated, but I'm just now realizing that the template isn't updated properly so the version isn't latest major - I'll fix that in a bit.

> Also why are you using both eslint and prettier?

Didn't change the template but I don't use prettier, so I'll prob remove those!

**Comment by mattxwang** [comment]
Oh forgot to mention - if this project really picks up speed, prettier _is_ useful for JSON, Markdown, etc. formatting too - but for now that seems unlikely 👀

**Comment by TimothyGu** [comment]
> Also why are you using both eslint and prettier?

FYI Prettier is generally considered a better formatter, as it destroys most formatting quirks in the original file, while ESLint respects the original code style more. Making code formatting as mechanical as possible is good because it [prevents disagreements](https://abseil.io/resources/swe_at_google.2.pdf#page=189). You can turn off ESLint's rules that deal with formatting with https://github.com/prettier/eslint-config-prettier.

**Comment by BryanPan342** [comment]
@TimothyGu

So to summarize?
- prettier => formating
- eslint => anything other than formating? (dependency checking, ordering, and other semantic issues etc.)

I honestly am not a big fan of precommit hooks that autoformat files, and tbh I haven't run into that any problems with different coding styles. But @mattxwang up to u if you want to add prettier!

**Comment by TimothyGu** [comment]
> So to summarize?
> 
> * prettier => formating
> * eslint => anything other than formating? (dependency checking, ordering, and other semantic issues etc.)

Yes, that is my personal preference. However, this is not my project, so it's up you and Matt which way you'd like to do it!

> I honestly am not a big fan of precommit hooks that autoformat files, and tbh I haven't run into that any problems with different coding styles.

I've personally gotten used to editor features (e.g., "file watchers") that automatically format the file as soon as I save. Problem generally arises when more people start working on the project, like folks with relatively less experience programming and who haven't gotten used to proper indentation, or people with significantly different experience from your own. (For instance, when I first started writing JavaScript in high school, I adhered to the [npm style guide](https://web.archive.org/web/20200723064147/https://docs.npmjs.com/misc/coding-style). It probably looks quite foreign to you.)

For a one-off project, obviously it doesn't matter too much. But it looks to me like this is a project intended to stand the test of time, and so a formatter (either ESLint or Prettier) would IMO be useful :)

**Comment by mattxwang** [comment]
Great points, and honestly just something I didn't really think about! Maybe I jumped the gun on removing prettier, I see no qualms with reintroducing it 👀 

Tim's comment did remind me - on other bigger open-source projects I've worked on, like [stylelint](https://github.com/stylelint/stylelint), they use prettier and eslint in tandem. 

> However, this is not my project

It's all of ours' project 😉 

(but seriously, I'll tell you about this more in-depth some time)

**Pull request merged**